bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -449,20 +455,62 @@ private static boolean isConvertibleToArrayType(Object sourceValue, BArrayType t
                 return true;
             }
         }
-        Set<Type> convertibleTypes;
+
         for (int i = 0; i < source.size(); i++) {
-            convertibleTypes = getConvertibleTypes(source.get(i), targetTypeElementType, unresolvedValues);
-            if (convertibleTypes.isEmpty()) {
+            if (!isConvertibleToArrayInstance(source.get(i), targetTypeElementType, unresolvedValues)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static boolean isConvertibleToTupleType(Object sourceValue, BTupleType targetType,
+                                                    List<TypeValuePair> unresolvedValues) {
+        if (!(sourceValue instanceof ArrayValue)) {
```

Comment:
Do we need this line. I think it will be always ArrayValue

Reply:
When the target type is tuple type, the execution will reach to this place and we can't guarantee the source type here is always an instance of `ArrayValue`. This check will ensure the success of `Object` to `ArrayValue` casting. So, IMO we need this check here.